code/video_fusion_player/src/stitcher.cpp

```diff
@@ -0,0 +1,387 @@
+#include "stitcher.h"
+
+//************************************
+// Method:    avframeToCvmat
+// Access:    public
+// Returns:   cv::Mat
+// Qualifier:
+// Parameter: const AVFrame * frame
+// Description: AVFrame转MAT
+//************************************
+cv::Mat avframeToCvmat(const AVFrame *frame) {
+	if (!frame) {
+        std::cerr << "Frame is null." << std::endl;
+        return cv::Mat(); // 返回空 Mat
+    }
+
+    int image_width = frame->width;
+    int image_height = frame->height;
+
+    // 创建 OpenCV Mat 对象
+    cv::Mat resMat(image_height, image_width, CV_8UC3);
+
+    // 创建转换上下文
+    SwsContext* avFrameToOpenCVBGRSwsContext = sws_getContext(
+        image_width,
+        image_height,
+        (AVPixelFormat)frame->format, 
+        image_width,
+        image_height,
+        AV_PIX_FMT_BGR24,
+        SWS_FAST_BILINEAR,
+        nullptr, nullptr, nullptr
+    );
+
+    if (!avFrameToOpenCVBGRSwsContext) {
+        std::cerr << "Could not create sws context." << std::endl;
+        return cv::Mat();
+    }
+
+    // 使用 sws_scale 转换图像
+    const uint8_t *srcSlice[1] = { frame->data[0] }; 
```

Comment:
There is one problem obviously. The AVFrame data maybe stored in multiple planes. For example, YUV420p has Y, U, and V.